test/e2e/mnist.py

```diff
@@ -176,13 +176,12 @@ def test_dataloader(self):
 
 # Initialize a trainer
 trainer = Trainer(
-    accelerator="auto",
+    accelerator="gpu",
```

Comment:
Can that be made conditional?

Reply:
Thinking if default value should be GPU or CPU

Reply:
Maybe something like `'gpu' if torch.cuda.is_available() else 'cpu'`?

Reply:
My intention is to have assurance of what accelerator is used.
So in case GPU isn't provisioned for some reason the GPU test should fail, to indicate that GPU wasn't tested. By using condition above the test will hide the fact that GPU wasn't tested.

Reply:
By providing external configuration (i.e. env variable, or splitting tests to run both CPU and GPU variants) I can make sure what is tested.

Reply:
I see, maybe an assertion somewhere else than the training script? Like in the test method?

Reply:
Or an environment variable?

Reply:
Ah we were typing at the same time and I didn't see you last suggestions :)

Yes, an environment variable looks like a good idea.

Reply:
Providing accelerator using env variable.